resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -142,15 +146,10 @@ private[spark] class Client(
         logError("Please check \"kubectl auth can-i create pod\" first. It should be yes.")
         throw e
     }
-    try {
-      val otherKubernetesResources = resolvedDriverSpec.driverKubernetesResources ++ Seq(configMap)
-      addOwnerReference(createdDriverPod, otherKubernetesResources)
-      kubernetesClient.resourceList(otherKubernetesResources: _*).createOrReplace()
-    } catch {
-      case NonFatal(e) =>
-        kubernetesClient.pods().delete(createdDriverPod)
-        throw e
-    }
+
+    // setup resources after pod creation, and refresh all resources owner references
+    val driverKubernetesResources = resolvedDriverSpec.driverKubernetesResources ++ Seq(configMap)
+    createOrReplaceResource(kubernetesClient, driverKubernetesResources, createdDriverPod)
```

Comment:
Please don't change this. We had better keep the original logic using `addOwnerReference`.